### PR TITLE
23 user can see saved jobs

### PR DESCRIPTION
**Pull request by NicoleKMarino**
A user can now save a job for later, view their saved jobs from their dashboard, delete a saved job, and can't save a job twice 

@jmejia- any suggestions for refactoring?

### CODE REVIEW

File: spec/features/employer/employer_can_login_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
```
Comment (roscalabrin) [inline spec/features/employer/employer_can_login_spec.rb:1]: @jmejia: I think we should add a test for multitenancy - i.e. a user can't see/access the employer dashboard. Does that make sense?

File: spec/features/saved_jobs/user_can_save_job_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    user = User.create(username: "lane", email: "lane@nil.com", password: "password")
+    visit login_path
+
+    fill_in "Username", with: "lane"
+    fill_in "Password", with: "password"
+    click_button "Login"
+
+    visit jobs_path
+
+    click_button("Save For Later")
```
Comment (roscalabrin) [inline spec/features/saved_jobs/user_can_save_job_spec.rb:16]: @NicoleKMarino we can add an assertion that to test that the count of saved jobs has incremented by one.:
it { should change { SavedJobs.count }.by 1 }

File: spec/features/user/user_can_have_role_spec.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
```
Comment (roscalabrin) [inline spec/features/user/user_can_have_role_spec.rb:2]: @jmejia: should this be a model test? I'm thinking that the fact that a user has a role is not a functionality.

File: spec/features/user/user_creation_spec.rb [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
+    fill_in "City", with: "Denver"
     fill_in "Password", with: "password"
+
     click_button "Create Account"
 
-    expect(page).to have_content "Logged in as lane"
     expect(page).to have_link "Logout"
-    expect(page).to have_link "Cart"
-    
+
```
Comment (roscalabrin) [inline spec/features/user/user_creation_spec.rb:16]: @NicoleKMarino we can also add here the assertion to check that the User tables count has increased by 1 (as I commented above)

File: spec/models/saved_job_spec.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe SavedJob, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (roscalabrin) [inline spec/models/saved_job_spec.rb:4]: @NicoleKMarino we're missing this test. We can test here the relationships between SavedJob and User and Job

File: app/controllers/saved_jobs_controller.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+class SavedJobsController < ApplicationController
+
+  def create
+    job = Job.find(params[:job_id])
```
Comment (roscalabrin) [inline app/controllers/saved_jobs_controller.rb:4]: @NicoleKMarino: we can refactor this:
pull out job = Job.find(params[:job_id]) into a new method, and call this method from create and destroy

File: spec/features/employer/employer_can_login_spec.rb (hunk 1 above)
Comment (jmejia) [inline spec/features/employer/employer_can_login_spec.rb:1]: @roscalabrin Yup, makes sense. Tests surrounding security and privacy are really important. It's common that if you are going to choose to _not_ test something that you will choose to cut out sad paths. However, this is a sad path that could have severe consequences. It's also highly likely that someone will try to exploit it so testing it is really important.

File: spec/features/user/user_can_have_role_spec.rb (hunk 3 above)
Comment (jmejia) [inline spec/features/user/user_can_have_role_spec.rb:2]: @roscalabrin Yup. We're not using Capybara (e.g. visiting a page) so no need to make it a feature test.

File: spec/features/employer/employer_can_login_spec.rb (hunk 1 above)
Comment (roscalabrin) [inline spec/features/employer/employer_can_login_spec.rb:1]: ^^^ @NicoleKMarino

File: spec/features/user/user_can_have_role_spec.rb (hunk 3 above)
Comment (roscalabrin) [inline spec/features/user/user_can_have_role_spec.rb:2]: ^^^ @NicoleKMarino

File: config/routes.rb [hunk 7]
```diff
@@ -6,12 +6,22 @@
...
 
-  resources :jobs_favorites, only: [:create, :update, :destroy]
+  namespace :employer do
+    resources :dashboard, only: [:index]
+    resources :jobs, only: [:new, :create, :edit, :update]
+  end
+
+
+  resources :companies, only: [:index]
+  get "/companies/:slug", to: "companies#show", as: "company"
```
Comment (jmejia) [inline config/routes.rb:16]: @NicoleKMarino Let's simplify these two lines with something like:

``` ruby
resources :companies, only: [:index, :show], param: :slug
```

Comment (jmejia) [inline config/routes.rb:16]: ^^ Unless there's a reason that won't work??

File: db/schema.rb [hunk 8]
```diff
@@ -24,7 +24,7 @@
     t.integer  "status",      default: 0
     t.text     "img_path"
     t.integer  "industry_id"
-    t.text     "slug"
+    t.string   "slug"
```
Comment (jmejia) [inline db/schema.rb:27]: Interesting change... Was there a problem with case insensitivity or something?

File: db/schema.rb [hunk 9]
```diff
@@ -64,6 +64,30 @@
...
+    t.datetime "updated_at", null: false
+  end
+
+  create_table "saved_jobs", force: :cascade do |t|
+    t.integer  "user_id"
+    t.integer  "job_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["job_id"], name: "index_saved_jobs_on_job_id", using: :btree
+    t.index ["user_id"], name: "index_saved_jobs_on_user_id", using: :btree
```
Comment (jmejia) [inline db/schema.rb:79]: Great use of indexing 👍

File: spec/features/admin/admin_login_spec.rb [hunk 10]
```diff
@@ -1,15 +1,24 @@
 require 'rails_helper'
 
 RSpec.feature "Admin can login" do
-  xscenario "registered admin can login" do
-    admin = create(:admin)
+  scenario "registered admin can login" do
+    role1= Role.create(name: "platform_admin")
+    user1= User.create(username: "Admin", email: "lanerdoce@aol.com", password: "password", first_name: "Lane", last_name: "Winham", address: "2122 Concord Lane", city: "Denver", state: "CO", zip_code: 80215, )
+    UserRole.create(user_id:user1.id, role_id:role1.id)
```
Comment (jmejia) [inline spec/features/admin/admin_login_spec.rb:7]: We can take advantage of ActiveRecord associations by doing it this way:

``` ruby
UserRole.create(user: user1, role: role1)
```

File: spec/models/role_spec.rb [hunk 11]
```diff
@@ -0,0 +1,6 @@
+# require 'rails_helper'
+#
+# RSpec.describe Role, type: :model do
+#
+#
+# end
```
Comment (jmejia) [inline spec/models/role_spec.rb:6]: @NicoleKMarino Let's get rid of this file or add a meaningful test.

**Comment by jmejia** [comment]
@NicoleKMarino This is a serious pull request with a ton of work. Nice job 💪 

Made some comments. You all are making great use of github for communication. Keep it up!

File: spec/features/user/user_can_have_role_spec.rb (hunk 3 above)
Comment (NicoleKMarino) [inline spec/features/user/user_can_have_role_spec.rb:2]: This was a test that @deborahleehamel wrote that I just re-wrote to make it passing. I will move it before committing.

**Pull request closed without merging**